improvement: S3C-4312 encryption info in ObjectMDLocation.setDataLocation

jonathan-gramain opened the pull request:
Support setting encryption info in ObjectMDLocation with the method
setDataLocation(), used by backbeat to set the new target location
before writing metadata on the target.

jonathan-gramain committed:
improvement: S3C-4312 encryption info in ObjectMDLocation.setDataLocation()

Support setting encryption info in ObjectMDLocation with the method
setDataLocation(), used by backbeat to set the new target location
before writing metadata on the target.

bert-e commented:
# Hello jonathan-gramain,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/arsenal/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Incorrect Jira project



The Jira issue S3C-4312 specified in the source
branch name, does not belong to project `ARSN`.

tcarmet commented:
/bypass_jira_check

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.1/improvement/S3C-4312-backbeatEncryptionSupport` with contents from `improvement/S3C-4312-backbeatEncryptionSupport`
and `development/8.1`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.1/improvement/S3C-4312-backbeatEncryptionSupport origin/development/8.1
 $ git merge origin/improvement/S3C-4312-backbeatEncryptionSupport
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.1/improvement/S3C-4312-backbeatEncryptionSupport
```





*The following options are set:* **bypass_jira_check**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/S3C-4312-backbeatEncryptionSupport` into
`development/7.10`
* [w/8.1/improvement/S3C-4312-backbeatEncryptionSupport](https://github.com/scality/arsenal/compare/development/8.1...w/8.1/improvement/S3C-4312-backbeatEncryptionSupport) will be merged into `development/8.1` (pull request #1511)



The following branches will **NOT** be impacted:

* `development/6.4`
* `development/7.4`




*Follow* integration pull requests if you would like to be notified of
build statuses by email.







*The following options are set:* **bypass_jira_check**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers











*The following options are set:* **bypass_jira_check**

miniscruff commented on lib/models/ObjectMDLocation.js:
```diff
@@ -33,9 +33,25 @@ class ObjectMDLocation {
+     * @return {ObjectMDLocation} return this
+     */
     setDataLocation(location) {
         this._data.key = location.key;
         this._data.dataStoreName = location.dataStoreName;
+        if (location.cryptoScheme) {
+            this._data.cryptoScheme = location.cryptoScheme;
+            this._data.cipheredDataKey = location.cipheredDataKey;
+        }
```
Could you do this all with Object.assign?
```suggestion
        this._data = Object.assign(this._data, location);
```

bert-e commented:
# Conflict




A conflict has been raised during the update of
integration branch `w/8.1/improvement/S3C-4312-backbeatEncryptionSupport` with contents from `improvement/S3C-4312-backbeatEncryptionSupport`
and `development/8.1`.

Please resolve the conflict on **the integration branch** (`w/8.1/improvement/S3C-4312-backbeatEncryptionSupport`).


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout w/8.1/improvement/S3C-4312-backbeatEncryptionSupport
 $ git pull  # or "git reset --hard origin/w/8.1/improvement/S3C-4312-backbeatEncryptionSupport"
 $ git merge origin/development/8.1
 $ # <intense conflict resolution>
 $ git commit
 $ git merge origin/improvement/S3C-4312-backbeatEncryptionSupport
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.1/improvement/S3C-4312-backbeatEncryptionSupport
```





*The following options are set:* **bypass_jira_check**

bert-e commented:
# History mismatch



Merge commit #f8405f74a45d971c58258d4f0a846c0604497be5 on the integration branch
`w/8.1/improvement/S3C-4312-backbeatEncryptionSupport` is merging a branch which is neither the current
branch `improvement/S3C-4312-backbeatEncryptionSupport` nor the development branch
`development/8.1`.

It is likely due to a rebase of the branch `improvement/S3C-4312-backbeatEncryptionSupport` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **bypass_jira_check**

jonathan-gramain commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **bypass_jira_check**

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.1/improvement/S3C-4312-backbeatEncryptionSupport` with contents from `improvement/S3C-4312-backbeatEncryptionSupport`
and `development/8.1`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.1/improvement/S3C-4312-backbeatEncryptionSupport origin/development/8.1
 $ git merge origin/improvement/S3C-4312-backbeatEncryptionSupport
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.1/improvement/S3C-4312-backbeatEncryptionSupport
```





*The following options are set:* **bypass_jira_check**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/S3C-4312-backbeatEncryptionSupport` into
`development/7.10`
* [w/8.1/improvement/S3C-4312-backbeatEncryptionSupport](https://github.com/scality/arsenal/compare/development/8.1...w/8.1/improvement/S3C-4312-backbeatEncryptionSupport) will be merged into `development/8.1` (pull request #1513)



The following branches will **NOT** be impacted:

* `development/6.4`
* `development/7.4`




*Follow* integration pull requests if you would like to be notified of
build statuses by email.







*The following options are set:* **bypass_jira_check**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers











*The following options are set:* **bypass_jira_check**

jonathan-gramain commented:
/approve

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/Arsenal/commit/2d539b5d87aeaf36f69594c055a84defdedaba75) did not succeed in branch w/8.1/improvement/S3C-4312-backbeatEncryptionSupport.




*The following options are set:* **approve, bypass_jira_check**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers











*The following options are set:* **bypass_jira_check**

jonathan-gramain commented on lib/models/ObjectMDLocation.js:
```diff
@@ -33,9 +33,20 @@ class ObjectMDLocation {
     setDataLocation(location) {
-        this._data.key = location.key;
-        this._data.dataStoreName = location.dataStoreName;
+        this._data = Object.assign(this._data, location);
```
@miniscruff I realized that neither the old code nor the change to use `Object.assign` are correct: indeed, we need to *overwrite* any encryption field with the one provided in the new location, i.e. the difference is that if the encryption params do not exist in the `location` parameter, they must be erased from `this._data.` I saw this in the Zenko tests that contain ObjectMDLocation tests, I'm considering backporting them to this 7.10 branch as well.

miniscruff replied:
oh ok, that makes sense. Will re-review when it is ready.

jonathan-gramain committed:
improvement: S3C-4312 backport + adapt ObjectMDLocation unit test

Backport and adapt to 7.x branch the ObjectMDLocation unit tests from
development/8.1 branch

bert-e commented:
# History mismatch



Merge commit #9227c56f349a1e989f02c9412af8359e27d30a27 on the integration branch
`w/8.1/improvement/S3C-4312-backbeatEncryptionSupport` is merging a branch which is neither the current
branch `improvement/S3C-4312-backbeatEncryptionSupport` nor the development branch
`development/8.1`.

It is likely due to a rebase of the branch `improvement/S3C-4312-backbeatEncryptionSupport` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **bypass_jira_check**

jonathan-gramain commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **bypass_jira_check**

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.1/improvement/S3C-4312-backbeatEncryptionSupport` with contents from `improvement/S3C-4312-backbeatEncryptionSupport`
and `development/8.1`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.1/improvement/S3C-4312-backbeatEncryptionSupport origin/development/8.1
 $ git merge origin/improvement/S3C-4312-backbeatEncryptionSupport
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.1/improvement/S3C-4312-backbeatEncryptionSupport
```





*The following options are set:* **bypass_jira_check**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/S3C-4312-backbeatEncryptionSupport` into
`development/7.10`
* [w/8.1/improvement/S3C-4312-backbeatEncryptionSupport](https://github.com/scality/arsenal/compare/development/8.1...w/8.1/improvement/S3C-4312-backbeatEncryptionSupport) will be merged into `development/8.1` (pull request #1514)



The following branches will **NOT** be impacted:

* `development/6.4`
* `development/7.4`




*Follow* integration pull requests if you would like to be notified of
build statuses by email.







*The following options are set:* **bypass_jira_check**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers











*The following options are set:* **bypass_jira_check**

miniscruff reviewed (approved):
Re-approving

jonathan-gramain commented:
/approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/7.10`

* :heavy_check_mark: `development/8.1`



The following branches will **NOT** be impacted:

* `development/6.4`
* `development/7.4`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve, bypass_jira_check**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/7.10`

* :heavy_check_mark: `development/8.1`



The following branches have **NOT** changed:

* `development/6.4`
* `development/7.4`



Please check the status of the associated issue S3C-4312.

Goodbye jonathan-gramain.
